contract/contracts/Voting/Voting.sol

```diff
@@ -30,34 +30,45 @@ contract Voting is OwnableUpgradeable, ReentrancyGuardUpgradeable, PausableUpgra
         __Pausable_init();
   }
 
-    function addProposal(Types.Proposal calldata _proposal) public {
-        require(bytes(_proposal.title).length > 0, "Title cannot be empty");
-
-        Types.Proposal storage newProposal = proposalsById[proposalCount];
+    function addProposal(
+            string memory _title,
+            string memory _description,
+            string memory _draftedBy,
+            bool _isElection,
+            uint256 _winnerCount,
+            address[] memory _voters,
+            address[] memory _candidates
+        ) public {
+            require(bytes(_title).length > 0, "Title cannot be empty");
+
+            Types.Proposal storage newProposal = proposalsById[proposalCount];
+            newProposal.id = proposalCount;
+            newProposal.title = _title;
+            newProposal.description = _description;
+            newProposal.draftedBy = _draftedBy;
```

Comment:
maybe we can make `drafterBy` is the `msg.sender`

Reply:
Good point